### PR TITLE
Issue/missing owning cluster

### PR DESCRIPTION
**Pull request by joaquin-corchero**
UI changes:
- Make "Any" not a valid option for the team selection (it wasn't triggering the "required" validation)
- Make the team email to be auto-populated

Swagger changes:
- Added validation details to the deployment map schema

Server changes:
- Added server side partial validation for post and put methods on the deployment map controller.
- Added unit tests for controller and a provider.

UI tests:
- UI test to check for owning team and email to be required on load

### CODE REVIEW

File: client/app/configuration/deployment-maps/DeploymentMapTargetController.js [hunk 1]
```diff
@@ -282,6 +287,15 @@ angular.module('EnvironmentManager.configuration').controller('DeploymentMapTarg
...
     }
 
+    function getDefaultCluster() {
+      var owningCluster = teamstorageservice.get('');
+      return owningCluster === 'Any' ? '' : owningCluster;
+    }
+
+    function getDefaultClusterEmail(defaultCluster) {
+      return defaultCluster === '' ? '' : getClusterByName(defaultCluster).Value.GroupEmailAddress;
+    }
```
Comment (jake-cross) [inline client/app/configuration/deployment-maps/DeploymentMapTargetController.js]: getClusterByName will return an empty list if there is no matching value. Can we handle there being no `Value` here. Or better yet, wrap it up in a `getClusterEmailByName()`

File: server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    mapStubs['../../models/Environment'] = deps.environmentMock;
+    mapStubs['../../models/EnvironmentType'] = deps.environmentTypeMock;
+    mapStubs['../data-access/clusters'] = deps.clustersMock;
+    mapStubs['../data-access/services'] = deps.servicesDbMock;
+    var infrastructureConfigurationProvider = proxyquire('../../../modules/provisioning/infrastructureConfigurationProvider', mapStubs);
+
+    var target = infrastructureConfigurationProvider;
+
+    return target.get(input.environmentName, input.serviceName, input.serverRoleName);
+  }
```
Comment (jake-cross) [inline server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js]: Could these helper functions get shifter to the bottom, so the file shows the tests first.

File: server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+      var promise = getInstance();
+
+      return promise.catch(error => {
+        error.should.have.property('message', expectedError);
+      });
+    });
+
+    it("Should return a configuration object", () => {
+      setupDependencies();
```
Comment (jake-cross) [inline server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js]: setupDependencies can be moved to a beforeEach

File: client/app/configuration/deployment-maps/DeploymentMapTargetController.js (hunk 1 above)
Comment (joaquin-corchero) [reply client/app/configuration/deployment-maps/DeploymentMapTargetController.js]: 👍

File: server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js (hunk 2 above)
Comment (joaquin-corchero) [reply server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js]: 👍

File: server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js (hunk 3 above)
Comment (joaquin-corchero) [reply server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js]: 👍

**Pull request merged**